go/vt/orchestrator/inst/instance_dao.go

```diff
@@ -1120,8 +1095,9 @@ func ReadClusterPrimary(clusterName string) ([](*Instance), error) {
 	condition := `
 		cluster_name = ?
 		and (replication_depth = 0 or is_co_primary)
+		and tablet_type = ?
```

Comment:
This additional step is required because earlier when we started a new tablet, it used to come up and its cluster_name would be set to the `hostname:port` of itself. But now, it will be set the keyspace/shard and then that new tablet used to show up in this `ReadClusterPrimary` function.

A better way is to also employ filtering on the tablet_type to be Primary.

Currently this function isn't unit tested, but it is end-to-end tested in `GracefulPrimaryTakeover` tests. I intend to add unit tests for all these functions that are reading data from the backend, but it will require some testing code setup, so I have deferred it to another PR.